clients/src/main/java/org/apache/kafka/clients/producer/KafkaProducer.java

```diff
@@ -1465,13 +1465,15 @@ public boolean isDone() {
     private class AppendCallbacks<K, V> implements RecordAccumulator.AppendCallbacks {
         private final Callback userCallback;
         private final ProducerInterceptors<K, V> interceptors;
-        private final ProducerRecord<K, V> record;
+        private ProducerRecord<K, V> record;
```

Comment:
Should we make `record` volatile since it's being updated and read by different threads?

Reply:
Will do.  And partition then should be volatile as well, then.